Armbian Build Framework Script [DOC-1]

Tearran opened the pull request:
## Pull Request Description

### Documentation-Focused: Armbian Build Framework 
#### Overview
This pull request introduces a user-friendly script that acts as a bridge to interact with the Armbian Build Framework. Its main goal is to enhance accessibility and facilitate communication for users who may not be familiar with the underlying script.

#### Key Features
- Provides a simplified interface for easier interaction.
- Improves documentation with enhanced help messages.
~- Supports multiple system languages (English, German, Spanish).~ todo
- JSON output.
- HTML output.
- Markdown output.
- bash menu output
- whiptail/dialog output

#### Future Enhancements
- Implementation of man pages generation.

#### Testing
Extensive testing will be conducted to ensure accurate and complete documentation generation, as well as fine-tuning of output formats for optimal readability.

#### Additional Information
The primary focus of this implementation is to enhance the accessibility and comprehensibility of the Armbian Build Framework through robust documentation.

Tearran committed:
Add files via upload

## Pull Request Description

### Documentation-Centric: Armbian Build Framework Wrapper Script

#### Overview
This pull request introduces a wrapper script designed with a strong emphasis on generating comprehensive and user-friendly documentation for the Armbian Build Framework. The implementation aims to provide various output formats, including man pages, markdown, and potentially JSON, to cater to different aspects of the larger system.

#### Features Implemented
- Added a basic wrapper script to provide a simplified interface.
- Improved help messages for better documentation.
- Initial support for multiple system languages (English, German, Spanish).
#### Features to be Implemented
- Implemented functionality to generate man pages for the wrapper script.
- Integrated support for generating markdown documentation.
- Designed with flexibility to potentially output JSON documentation.

#### Continuinf Testing
- Rigorous testing of the documentation generation process to ensure accuracy and completeness.
- Fine-tuning of the output formats for optimal readability.

#### Additional Notes
- The primary focus of this implementation is to enhance the accessibility and comprehensibility of the Armbian Build Framework through robust documentation.

Tearran committed:
Update armbian-build.sh

added json

Tearran committed:
Update armbian-build.sh

Tearran committed:
Update armbian-build.sh

fix error

Tearran committed:
Update armbian-build.sh

implemented Translation switch in all function

adeepn commented:
You are doing something strange with localization)

https://mywiki.wooledge.org/BashFAQ/098

Tearran committed:
Update armbian-build.sh

drooped example functions to fix language switching. json output seems working

Tearran commented:
> You are doing something strange with localization)
> 
> https://mywiki.wooledge.org/BashFAQ/098

@adeepn It seems like you found something interesting with localization! I appreciate the link to the BashFAQ, and running LANGUAGE=es_ES ./armbian-build.sh was quite insightful.

I want to mention that the language selection in my script was primarily a placeholder for addressing locales. The main focus was on formatting the English output into JSON, key pairs, and dialog boxes. A loop for gettext could potentially be set up to improve localization.

It's a work in progress, and I'm open to any suggestions or improvements.

Thanks for your input!

Tearran committed:
Update and rename armbian-build.sh to armbian-build

removed language switch.
fixed loop iteration of descriptions

Tearran committed:
Update armbian-build

added tui output

Tearran committed:
Update armbian-build

Tearran committed:
Update armbian-build

Tearran commented:
## Quick Video showing the outputs
https://youtu.be/W1GXxkw-0zs

### Example of the html output
https://codepen.io/Tearran/pen/ZEwKaXe

## Quick use
```bash
{
wget https://raw.githubusercontent.com/Tearran/build/DOC-1/armbian-build ; 
bash ./armbian-build -h ;
}
```

Tearran committed:
Update and rename armbian-build to doc/armbian-build

| Feature |  desc | example | src_reference | status |

| :-----: | :---: | :---: | :-----: | :---------: |

quick formatting to array and a markdown table

Tearran commented:
Will output features.md and featuesr.sh 
for example 

# Hidden_options

| Feature |  desc | example | src_reference | status |
| :-----: | :---: | :---: | :-----: | :---------: |
| ARMBIAN_MIRROR | Overrides automated mirror selection | null | [references](null) |  active | 
| AUFS | Include support for AUFS | null | [references](null) |  active |

Tearran committed:
Update and rename armbian-build to doc.sh

leggewie reviewed (changes requested):
please rebase and squash those twelve commits into relevant, atomic commits.

please have a look through AI review and either dismiss (with good reason) or kindly address the issue raised.

am I correct to think that the data for this script will still have to be provided one way or another?  For example via #6389 ?
